app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt

```diff
@@ -25,47 +32,63 @@ class MainActivity : AppCompatActivity() {
-            .build()
 
-        val api = retrofit.create(ApiService::class.java)
+        // Create an instance of the tab layout from the view.
+        val tabLayout = findViewById<TabLayout>(R.id.tab_layout)
+        // Set the text for each tab.
+        for(title in CategoryUtils.categoryFromTabIndex){
+            tabLayout.addTab(tabLayout.newTab().setText(title.capitalize()))
+        }
+        // Set the tabs to fill the entire layout.
+        tabLayout.tabGravity = TabLayout.GRAVITY_FILL
 
-        api.fetchAllArticles().enqueue(object: Callback<FetchResponse>{
-            override fun onResponse(call: Call<FetchResponse>, response: Response<FetchResponse>) {
-                showArticles(response.body()!!.articles)
-            }
+        // Use PagerAdapter to manage page views in fragments.
+        // Each page is represented by its own fragment.
+        val viewPager = findViewById<ViewPager>(R.id.pager)
+
+        var adapter: TabsPagerAdapter
+        adapter = TabsPagerAdapter(supportFragmentManager, tabLayout.tabCount)
 
-            override fun onFailure(call: Call<FetchResponse>, t: Throwable) {
-                Timber.d(t.toString()) // Log the failure.
+        viewPager.adapter = adapter
+
+        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))
+
+        tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
+            override fun onTabSelected(tab: TabLayout.Tab) {
+                viewPager.currentItem = tab.position
             }
-        })
-    }
 
-    private fun showArticles(articles: List<Article>) {
-        val itemOnClick: (View, Int, Int) -> Unit = { _, position, _ ->
-            // The URL belonging to the selected article.
-            val articleUrl = articles[position].url
-            // Initialize the Custom Tabs Intent Builder.
-            val ctBuilder: CustomTabsIntent.Builder = CustomTabsIntent.Builder()
-            // Show the website's title in the Custom Tab.
-            ctBuilder.setShowTitle(true)
-            // Set the toolbar's color to be the app's primary color.
-            ctBuilder.setToolbarColor(ResourcesCompat.getColor(resources,
-                R.color.colorPrimary, null))
-            // Build the Custom Tabs intent.
-            val ctIntent: CustomTabsIntent = ctBuilder.build()
-            // Parse the article URL as a Uri, and then launch the
-            // intent.
-            ctIntent.launchUrl(this@MainActivity, Uri.parse(articleUrl))
-        }
+            override fun onTabUnselected(tab: TabLayout.Tab) {}
 
-        recyclerView.apply {
-            layoutManager = LinearLayoutManager(this@MainActivity)
-            adapter =
-                ArticlesAdapter(this@MainActivity, articles, itemOnClick)
-        }
+            override fun onTabReselected(tab: TabLayout.Tab) {}
+        })
     }
+
+//    private fun showArticles(articles: List<Article>) {
```

Comment:
Can remove all this commented code?? LOL can you clean up ur PRs before getting them reviewed !!